site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5158,14 +5158,95 @@ public function getNumberGradeInquiries($gradeable_id, $is_grade_inquiry_per_com
+                GROUP BY gcd.gcd_grader_id
+            ", [$gradeable_id]);
+            foreach ($this->course_db->rows() as $row) {
+                $return[$row['gcd_grader_id']] = $row['count'];
+            }
+            return $return;
+        }
+        else {
+            $this->course_db->query("
+                SELECT
+                    count(result.*),
+                    result.gcd_grader_id
+                FROM (
+                    SELECT
+                        DISTINCT gi.user_id,
+                        gcd.gcd_grader_id
+                    FROM grade_inquiries gi
+                    INNER JOIN gradeable_data gd ON (
+                        gi.g_id = gd.g_id
+                        AND gi.user_id = gd.gd_user_id
+                    )
+                    INNER JOIN gradeable_component_data gcd ON (
+                        gd.gd_id = gcd.gd_id
+                    )
+                    WHERE 
+                        gi.status = -1 
+                        AND gi.g_id = ?
+                ) AS result
+                GROUP BY result.gcd_grader_id
+            ", [$gradeable_id]);
+            foreach ($this->course_db->rows() as $row) {
+                $return[$row['gcd_grader_id']] = $row['count'];
+            }
+            return $return;
+        }
+    }
+
     /*
-     * This is used to convert one of the by component inquiries per student for a gradeable to a non-component inquiry.
+     * This is used to convert one of the by-component inquiries per student for a gradeable to a non-component inquiry.
      * This allows graders to still respond to by component inquiries if in no-component mode.
      */
     public function convertInquiryComponentId($gradeable) {
         $this->course_db->query("UPDATE grade_inquiries SET gc_id=NULL WHERE id IN (SELECT a.id FROM (SELECT DISTINCT ON (t.user_id) user_id, t.id FROM (SELECT * FROM grade_inquiries ORDER BY id) t WHERE t.g_id=?) a);", [$gradeable->getId()]);
     }
 
+    /**
+     * This is used to revert non-component inquiries back to by-component inquiries
+     * convertInquiryComponentId function modifies the lowest gc_id of each student of the gradeable to null.
+     * If instructor decides to switch to non-component from by-component then switch back to by-component,
+     * this revertInquiryComponentId function will fetch the gc_id from grade_inquiry_discussion and update accordingly.
+     * @param Gradeable $gradeable
+     */
+    public function revertInquiryComponentId(Gradeable $gradeable): void {
+        $this->course_db->query("
+            UPDATE grade_inquiries
+            SET gc_id = (
+                SELECT gid.gc_id
+                FROM grade_inquiry_discussion gid
+                WHERE grade_inquiries.id = gid.grade_inquiry_id
+            )
+            WHERE
+                grade_inquiries.gc_id IS NULL
+                AND grade_inquiries.g_id = ?
+        ", [$gradeable->getId()]);
+    }
```

Comment:
It feels like there might be edge cases we're missing here, but I can't quite figure out why it feels wrong.

Reply:
After some thoughts, I agree that there are some corner cases that is not handled. One case is submissions made during non-component setting. In such case grade_inquiry_discussion gc_id is null as well, so there is no point of updating the grade_inquiries table (b/c both gc_id are null). Hence, it works only for submissions made during by_component setting. However, I tested locally and students submitted during non-component will continue to see All inquiry discussion, so it should not be a huge problem.

Reply:
Discussed with @ziesski 
This is actually building on a bug/oversight in the original code.
When we switch from 'by component' to 'not by component' ALL gc_ids in the inquiry should be cleaned up and made NULL.  This function should not exist.  We should not be attempting to revive stale gc_id data in these super rare 'what if' scenario.  This function should be simplified/deleted when this bug is removed.  

@ziesski is making a separate issue and will like this PR and this function.